ui/src/components/issuers.tsx/IssuersTable.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  Avatar,
+  Button,
+  Card,
+  Radio,
+  RadioChangeEvent,
+  Row,
+  Space,
+  Table,
+  TableColumnsType,
+  Typography,
+} from "antd";
+import { useCallback, useEffect, useState } from "react";
+import { useSearchParams } from "react-router-dom";
+import { ErrorResult } from "../shared/ErrorResult";
+import { NoResults } from "../shared/NoResults";
+import { TableCard } from "../shared/TableCard";
+import { identifierParser } from "src/adapters/api/issuers";
+import IconIssuers from "src/assets/icons/building.svg?react";
+import IconPlus from "src/assets/icons/plus.svg?react";
+import { useIssuerContext } from "src/contexts/Issuer";
+import { Issuer } from "src/domain/identifier";
+import { isAsyncTaskDataAvailable, isAsyncTaskStarting } from "src/utils/async";
+
+import { QUERY_SEARCH_PARAM } from "src/utils/constants";
+
+export function IssuersTable({ handleAddIssuer }: { handleAddIssuer: () => void }) {
+  const { handleChange, identifier, issuersList } = useIssuerContext();
+  const [filteredIdentifiers, setFilteredIdentifiers] = useState<Issuer[]>(() =>
+    isAsyncTaskDataAvailable(issuersList) ? issuersList.data : []
+  );
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const queryParam = searchParams.get(QUERY_SEARCH_PARAM);
+
+  const handleIssuerChange = ({ target: { value } }: RadioChangeEvent) => {
+    const parsedIdentifier = identifierParser.safeParse(value);
+
+    if (parsedIdentifier.success) {
+      handleChange(parsedIdentifier.data);
+    } else {
+      handleChange(null);
+    }
+  };
+
+  useEffect(() => {
+    if (isAsyncTaskDataAvailable(issuersList)) {
+      if (queryParam === null) {
+        setFilteredIdentifiers(issuersList.data);
+      } else {
+        const filteredData = issuersList.data.filter((issuer: Issuer) =>
+          Object.values(issuer).some((value: string) => value.includes(queryParam))
+        );
+        setFilteredIdentifiers(filteredData);
+      }
+    }
+  }, [queryParam, issuersList]);
+
+  const onSearch = useCallback(
+    (query: string) => {
+      setSearchParams((previousParams) => {
+        const previousQuery = previousParams.get(QUERY_SEARCH_PARAM);
+        const params = new URLSearchParams(previousParams);
+
+        if (query === "") {
```

Comment:
`if (!query) {`

Reply:
<img width="365" alt="Screenshot 2024-08-07 at 20 04 37" src="https://github.com/user-attachments/assets/5c753516-49ae-4ee7-b836-f3928f5a27d7">

I thought this check made sense with an empty string, so I did it the same way as in other components.